--- cv_scaler_org.cc	2018-11-08 12:20:45.946571912 +0100
+++ cv_scaler.cc	2018-11-08 12:19:35.032859600 +0100
@@ -194,7 +194,8 @@
   note += adc_lp_[ADC_CHANNEL_CV_V_OCT] * calibration_data_->pitch_scale;
   
   performance_state->note = note;
-  performance_state->tonic = 12.0f + transpose_;
+  //performance_state->tonic = 12.0f + transpose_;
+  performance_state->tonic = 24.0f + (round(transpose_/12.0f)*12.0f);
   
   DetectNormalization();
   
@@ -210,7 +211,8 @@
   if (performance_state->internal_note) {
     // Remove quantization when nothing is plugged in the V/OCT input.
     performance_state->note = 0.0f;
-    performance_state->tonic = 12.0f + transpose;
+    //performance_state->tonic = 12.0f + transpose;
+    performance_state->tonic = 24.0f + transpose_;
   }
   
   // Hysteresis on chord.
